tensorflow_encrypted/protocol/securenn.py

```diff
@@ -106,3 +115,44 @@ def max_pool(self, x):
 
     def dmax_pool_efficient(self, x):
         raise NotImplementedError
+
+
+def _lsb_private(prot: SecureNN, y: PondPrivateTensor):
+    with tf.name_scope('lsb_mask'):
+        with tf.device(prot.crypto_producer.device_name):
+            x = prot.tensor_factory.Tensor.sample_uniform(y.shape)
+            xbits = x.binarize()
+            xlsb = xbits[..., 0]
+            x = PondPrivateTensor(prot, *prot._share(x), is_scaled=True)
+            xbits = PondPrivateTensor(prot, *prot._share(xbits), is_scaled=False)
+            xlsb = PondPrivateTensor(prot, *prot._share(xlsb, p), is_scaled=False)
+            # TODO: Generate zero mask?
+
+        devices = [prot.server0.device_name, prot.server1.device_name]
+        bits_device = random.choice(devices)
+        with tf.device(bits_device):
+            b = _generate_random_bits(y.shape)
+            b0, b1 = b.unwrapped
+
+    with tf.name_scope('lsb'):
```

Comment:
I would move this name scope to the top to also include `lsb_mask` to avoid two nodes in the graph -- wdyt?

Reply:
sounds good -- perhaps then we'll want to switch this subscope to something like `lsb_ops`?, versus leaving it under `lsb`?

Reply:
don't think `masking` is completely off, could be no subscope as well; `ops` is a bit generic